resharper/resharper-yaml/src/Psi/Parsing/YamlLexerGenerated.cs

```diff
@@ -0,0 +1,343 @@
+      }
+    }
+
+    public TokenNodeType TokenType => LocateToken();
+
+    public int TokenStart
+    {
+      get
+      {
+        LocateToken();
+        return yy_buffer_start;
+      }
+    }
+
+    public int TokenEnd
+    {
+      get
+      {
+        LocateToken();
+        return yy_buffer_end;
+      }
+    }
+
+    public IBuffer Buffer => yy_buffer;
+    public uint LexerStateEx => (uint) yy_lexical_state;
+
+    public int EOFPos => yy_eof_pos;
+    public int LexemIndent => 7;  // No, I don't know why
+
+    private bool IsBlock => flowLevel == 0;
+
+    private TokenNodeType LocateToken()
+    {
+      if (currentTokenType == null)
+      {
+        try
+        {
+          currentTokenType = _locateToken();
+        }
+        catch (Exception e)
+        {
+          e.AddData("TokenType", () => currentTokenType);
+          e.AddData("LexerState", () => LexerStateEx);
+          e.AddData("TokenStart", () => yy_buffer_start);
+          e.AddData("TokenPos", () => yy_buffer_index);
+          e.AddData("Buffer", () =>
+          {
+            var start = Math.Max(0, yy_buffer_end);
+            var tokenText = yy_buffer.GetText(new TextRange(start, yy_buffer_index));
+            tokenText = Regex.Replace(tokenText, @"\p{Cc}", a => $"[{(byte) a.Value[0]:X2}]");
+            return tokenText;
+          });
+          throw;
+        }
+      }
+
+      return currentTokenType;
+    }
+
+    private void RewindToken()
+    {
+      yy_buffer_end = yy_buffer_index = yy_buffer_start;
+    }
+
+    private void RewindChar()
+    {
+      yy_buffer_end = yy_buffer_index = yy_buffer_index - 1;
+    }
+
+    private void RewindWhitespace()
+    {
+      while (yy_buffer_index > 0 && IsWhitespace(yy_buffer[yy_buffer_index - 1]))
+        yy_buffer_end = yy_buffer_index = yy_buffer_index - 1;
+    }
+
+    private static bool IsWhitespace(char c) => c == ' ' || c == '\t';
+
+    private void PushFlowIndicator()
+    {
+      flowLevel++;
+      parentNodeIndent = yy_buffer_start - lastNewLineOffset;
+      yybegin(FLOW);
+    }
+
+    private void PopFlowIndicator()
+    {
+      flowLevel = Math.Max(0, flowLevel - 1);
+      if (IsBlock)
+        yybegin(BLOCK);
+    }
+
+    private void ResetBlockFlowState()
+    {
+      yybegin(IsBlock ? BLOCK : FLOW);
+    }
+
+    private void HandleNewLine(bool resetParentNodeIndent = true)
+    {
+      currentLineIndent = 0;
+      if (resetParentNodeIndent)
+        parentNodeIndent = -1;
+      lastNewLineOffset = yy_buffer_end;
+    }
+
+    private TokenNodeType HandleImplicitKey()
+    {
+      parentNodeIndent = yy_buffer_start - lastNewLineOffset;
+
+      // We special case implicit keys as a single line ns-plain followed by a colon
+      // But we want to return that as separate tokens. So match it, rewind and let
+      // the next pass handle the whitespace and colon
+      // TODO: This doesn't reset line indents or last new line offset!!
```

Comment:
looks like important todo :)

Reply:
They're all important :) I agree, but I haven't hit it in any tests yet. I'll get round to it before RTM.